pootle/static/js/shared/components/AutoIcon.js

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) Zing contributors.
+ *
+ * This file is a part of the Zing project. It is distributed under the GPL3
+ * or later license. See the LICENSE file for a copy of the license and the
+ * AUTHORS file for copyright and authorship information.
+ */
+
+import React from 'react';
+
+import crc16 from '../utils/crc16'; // use relative path, otherwise tests will fail
+
+export function calcAbbreviation(title, length) {
+  const regexA = /\b[A-Z]/g;
+  const regexB = /\b[\w]/g;
+
+  let out = [];
+  let matches;
+  while ((matches = regexA.exec(title))) {
+    out.push(matches[0]);
+    if (out.length === length) {
+      return out.join('');
+    }
+  }
+
+  out = [];
+  while ((matches = regexB.exec(title))) {
+    out.push(matches[0]);
+    if (out.length === length) {
+      return out.join('').toUpperCase();
+    }
+  }
+
+  return title.substr(0, length).toUpperCase();
+}
+
+export function calcHue(title) {
+  return crc16(title) % 360;
+}
+
+class AutoIcon extends React.Component {
+
+  static propTypes() {
+    return {
+      title: React.PropTypes.string.isRequired,
+      saturation: React.PropTypes.number,
+      lightness: React.PropTypes.number,
+      abbreviationLength: React.PropTypes.number,
+      mode: React.PropTypes.string,
+      size: React.PropTypes.number,
+    };
+  }
+
+  constructor(props) {
+    super();
+
+    this.state = Object.assign({
```

Comment:
Using `Object.assign` directly will definitely break IE11. Unless we plan to stop supporting it in the near future, please use `assign` from the `object-assign` package instead (`import assign from 'object-assign';`).